tests/src/test/java/alluxio/client/cli/job/DistributedCommandsCancelStatsTest.java

```diff
@@ -0,0 +1,195 @@
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.cli.job;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import alluxio.cli.fs.FileSystemShell;
+import alluxio.client.file.FileSystemTestUtils;
+import alluxio.conf.PropertyKey;
+import alluxio.conf.ServerConfiguration;
+import alluxio.grpc.WritePType;
+import alluxio.job.plan.load.LoadConfig;
+import alluxio.job.plan.migrate.MigrateConfig;
+import alluxio.job.plan.persist.PersistConfig;
+import alluxio.job.util.JobTestUtils;
+import alluxio.job.wire.Status;
+import alluxio.master.LocalAlluxioJobCluster;
+import alluxio.metrics.MetricKey;
+import alluxio.metrics.MetricsSystem;
+import alluxio.testutils.LocalAlluxioClusterResource;
+
+import com.google.common.collect.Sets;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.Collections;
+
+public class DistributedCommandsCancelStatsTest extends JobShellTest {
+  @ClassRule
+  public static LocalAlluxioClusterResource sResource =
+      new LocalAlluxioClusterResource.Builder()
+          .setNumWorkers(4)
+          .setProperty(PropertyKey.MASTER_PERSISTENCE_CHECKER_INTERVAL_MS, "10ms")
+          .setProperty(PropertyKey.MASTER_PERSISTENCE_SCHEDULER_INTERVAL_MS, "10ms")
+          .setProperty(PropertyKey.JOB_MASTER_WORKER_HEARTBEAT_INTERVAL, "200ms")
+          .setProperty(PropertyKey.WORKER_RAMDISK_SIZE, SIZE_BYTES)
+          .setProperty(PropertyKey.USER_BLOCK_SIZE_BYTES_DEFAULT, SIZE_BYTES)
+          .setProperty(PropertyKey.MASTER_TTL_CHECKER_INTERVAL_MS, Integer.MAX_VALUE)
+          .setProperty(PropertyKey.USER_FILE_WRITE_TYPE_DEFAULT, "CACHE_THROUGH")
+          .setProperty(PropertyKey.USER_FILE_RESERVED_BYTES, SIZE_BYTES / 2)
+          .setProperty(PropertyKey.CONF_DYNAMIC_UPDATE_ENABLED, true)
+          .build();
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+
+    sLocalAlluxioCluster = sResource.get();
+    sLocalAlluxioJobCluster = new LocalAlluxioJobCluster();
+    sLocalAlluxioJobCluster.start();
+    sFileSystem = sLocalAlluxioCluster.getClient();
+    sJobMaster = sLocalAlluxioJobCluster.getMaster().getJobMaster();
+    sJobShell = new alluxio.cli.job.JobShell(ServerConfiguration.global());
+    sFsShell = new FileSystemShell(ServerConfiguration.global());
+  }
+
+  @Test
+  public void testDistributedLoadCancelStats() throws Exception {
+    FileSystemTestUtils.createByteFile(sFileSystem, "/testFileNew", WritePType.THROUGH, 10);
+
+    long jobId = sJobMaster.run(new LoadConfig("/testFileNew", 1, Collections.EMPTY_SET,
+            Collections.EMPTY_SET, Collections.EMPTY_SET, Collections.EMPTY_SET, false));
+
+    sJobShell.run("cancel", Long.toString(jobId));
```

Comment:
is this cancel operation guaranteed to work before the job is complete?  have the same concern about other unit tests.

Reply:
Yes, i think so, below in the test it will check if the job is cancelled or not, if cancels successfully then the test will pass, otherwise it will fail. Same for other tests.

Reply:
I think @jja725 's concern is that the cancel might be a noop if the job has already finished by the time it reaches the cancel call, since the job is very small.

Reply:
in that case, we are not testing anything at all.

Reply:
If the job has finished, then the output and stat counter values below would be different, and this test would fail. Do I interpret correctly?

Reply:
I see, then it depends on the timing of the test， which may lead to flakey test.  Please add a comment to the test indicating that some timing assumption is used in this test.

Reply:
Yes, it depends on timing of the test, will add such a comment.

Reply:
Done.